library/platform_util.c

```diff
@@ -119,43 +119,45 @@ void *mbedtls_platform_zeroize( void *buf, size_t len )
 
 void *mbedtls_platform_memset( void *ptr, int value, size_t num )
 {
-    size_t i, start_offset;
+    size_t i, start_offset = 0;
     volatile size_t flow_counter = 0;
     volatile char *b = ptr;
     char rnd_data;
+    if( num > 0 )
```

Comment:
if num == 0, what will be returned ?

Reply:
A good pointer, the operation will be successful. This cased is used and tested in Mbed TLS tests.

Reply:
Ok, missed that flow control checked against 2 * num and 0 is a success value :)